src/io/obf.rs

```diff
@@ -72,233 +72,179 @@ pub async fn obfuscate<M, SU, SH, ST, R, P>(
 
     Sample the initial public key (level 0) with our reveal flags.
 
-    Sample the initial BGG+ encodings.
-    **NOTE:** the paper treats
-        - c_att = encoding of (1, bits(X)), where X represents the evaluator's inputs.
-        - c_t = encoding of the FHE secret-key
-    as two distinct vectors.
-
-    The length of encodings_init is (1 + packed_input_size + 1):
-       - 1 for the encoding of 1
-       - packed_input_size for the packed evaluator inputs,
-       - 1 for the encoding of t (secret key).
-
-    encodings_init is bundled into a single Vec<bool> of length L,
-    where:
-       - indices 0..L-2 → c_att
-       - index   L-1   → c_t
+    The length of encodings_init is (1 + packed_input_size):
+       - 1 the encoding of t(secret key)
+       - packed_input_size for the packed evaluator inputs and for the encoding of 1.
     =============================================================================
     */
 
+    // Sample BGG+ encoding secret key.
+    // We sample multiple s_bar because we are using module LWE to reduce FFT burden by reducing n.
+    let s_bars = sampler_uniform.sample_uniform(&params, 1, d, DistType::BitDist).get_row(0);
+    log_mem("Sampled s_bars");
     // Sample FHE secret key t
-    let t_bar_matrix = sampler_uniform.sample_uniform(&params, 1, 1, DistType::FinRingDist);
-    log_mem("Sampled t_bar_matrix");
-    let minus_t_bar = -t_bar_matrix.entry(0, 0);
-    let mut plaintexts =
-        (0..(packed_input_size - 1)).map(|_| M::P::const_zero(&params)).collect_vec();
-    plaintexts.push(minus_t_bar.clone());
+    let t_bar = sampler_uniform.sample_uniform(&params, 1, 1, DistType::BitDist);
+    log_mem("Sampled t_bar");
+    // This is actually shorten version from paper where it defined t := (t_bar, -1), but instead
```

Comment:
This is not correct. In the paper `t` is defined as `(t_bar, -1)` because of the specific caracteristics of the GSW FHE scheme that is employed in the paper. In the scheme used here the secert key `t` is simply equal to `t_bar`. The fact that we pass `- t_bar` as input is just an unrelated optimization trick.